File: 1894/Solution.js
/**
 * @param {number[]} chalk
 * @param {number} k
 * @return {number}
 */
var chalkReplacer = function(chalk, k) {
    let sum = 0;
    for(let i = 0; i < chalk.length; i++){
        sum += chalk[i];
    }
    sum = k % sum;
    for(let i = 0; i < chalk.length; i++){
        sum -= chalk[i];
        if(sum < 0) {
            return i;
        }
    }
};

File: 1894/Solution.go
func chalkReplacer(chalk []int, k int) int {
    sum := 0
    for _, val := range chalk {
        sum += val
    }
    sum = k % sum
    for i, val := range chalk {
        sum -= val
        if sum < 0 {
            return i
        }
    }
    return -1
}
